src/main/java/com/aws/greengrass/clientdevices/auth/iot/IotAuthClient.java

```diff
@@ -41,13 +41,13 @@
 import javax.inject.Inject;
 
 public interface IotAuthClient {
-    Optional<String> getActiveCertificateId(String certificatePem);
+    Optional<String> getActiveCertificateId(String certificatePem) throws CloudServiceInteractionException;
```

Comment:
this breaks out interface doesn't it? Isn't moquette using these methods? If it is, then this is going to break consumers.

Reply:
consumers just interact with `ClientDevicesAuthServiceApi` and `AuthenticationException`s from what i can see, so that should be fine?